wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -372,9 +371,10 @@ public WorkspaceDto update(
       @ApiParam(value = "The workspace update", required = true) WorkspaceDto update)
       throws BadRequestException, ServerException, ForbiddenException, NotFoundException,
           ConflictException {
-    checkArgument(
-        update.getConfig() != null ^ update.getDevfile() != null,
-        "Required non-null workspace configuration or devfile update but not both");
+    if (update.getDevfile() != null) {
```

Comment:
I know this is meant to be a temporary fix but shouldn't this be handled on a slightly lower level than in the REST service?

Reply:
Now, REST is a level where we add stub workspace config for UI and then null it. And it's OK for me.